File: routes/user.routes.go
package routes

import (
	createUser "themoviebakery/controllers/user-controllers/create"
	updateUser "themoviebakery/controllers/user-controllers/update"
	deleteUserHandler "themoviebakery/handlers/user-handlers/delete"
	getUserHandler "themoviebakery/handlers/user-handlers/get"

	"github.com/gin-gonic/gin"
)

func InitUserRoutes(router *gin.Engine) {
	groupRoute := router.Group("/api/v1") // .Use(middleware.Auth())
	groupRoute.POST("/user", createUser.CreateUser)
	groupRoute.GET("/user-by-email", getUserHandler.HandlerGetUserByEmail)
	groupRoute.GET("/user-by-id", getUserHandler.HandlerGetUserById)
	groupRoute.PUT("/user", updateUser.UpdateUser)
	groupRoute.DELETE("/user", deleteUserHandler.DeleteUserById)
}

File: src/routes/user.routes.ts
import { Router } from 'express'
import UserController from '../controllers/user'

const userRouter = Router()
const userController = new UserController()

userRouter.post('/create', userController.create)
userRouter.post('/create-google', userController.createGoogle)
userRouter.get('/get-by-email', userController.getByEmail)
userRouter.get('/get-by-id', userController.getById)

export default userRouter
